Exeption on zpopmin

## Bug Report

<!-- First of all: Have you checked the docs https://github.com/lettuce-io/lettuce-core/wiki, GitHub issues, or Stack Overflow whether someone else has already reported your issue? -->

#### Current Behavior

<!-- A clear and concise description of the behavior.-->
<!-- See https://stackoverflow.com/help/minimal-reproducible-example for details of what a good minimum reproducible example is -->

I encoutered an exeption  while poping something from an sorted set.

<details>
<summary>Stack trace</summary>

```java
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: Caused by: java.lang.UnsupportedOperationException: io.lettuce.core.pubsub.PubSubCommandHandler$ResponseHeaderReplayOutput does not support set(double)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.output.CommandOutput.set(CommandOutput.java:123)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.RedisStateMachine.safeSet(RedisStateMachine.java:798)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.RedisStateMachine.handleFloat(RedisStateMachine.java:455)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.RedisStateMachine$State$Type.handle(RedisStateMachine.java:210)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.RedisStateMachine.doDecode(RedisStateMachine.java:363)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.RedisStateMachine.decode(RedisStateMachine.java:324)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:857)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.CommandHandler.decode0(CommandHandler.java:808)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:791)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:674)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.pubsub.PubSubCommandHandler.decode(PubSubCommandHandler.java:97)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at PrimeCorePlugin-0.3.7-SNAPSHOT.jar//io.lettuce.core.protocol.CommandHandler.channelRead(CommandHandler.java:614)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[01:25:58] [ForkJoinPool.commonPool-worker-5/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1583)

```

</details>

#### Input Code

<!-- Java/Kotlin/Scala/Groovy/… or Repo link if applicable: -->

<details>
<summary>Input Code</summary>

```kotlin
import io.lettuce.core.RedisClient
import io.lettuce.core.codec.RedisCodec
import io.lettuce.core.codec.StringCodec

fun main() {
    val client = RedisClient.create("redis://localhost")
    val connection = client.connectPubSub(RedisCodec.of(StringCodec.ASCII, StringCodec.UTF8)).sync()

    connection.zadd("myzset", 1.0, "one")
    connection.zadd("myzset", 2.0, "two")

    connection.zpopmin("myzset", 1).firstOrNull()?.takeIf { it.hasValue() }?.let {
        println("Popped value: $it")
    } ?: println("No value to pop")
}
```

</details>

#### Expected behavior/code

<!-- A clear and concise description of what you expected to happen (or code). -->
Well I expect it to not throw an exeption.

#### Environment

- Lettuce version(s): lettuce-core:6.5.4.RELEASE
- Redis version: 	redis:7.2@sha256:6da5f3c2579824a9de659e612242f3b11bc421f8aa847ba33e427c1bc122ed3d

#### Possible Solution

<!-- Only if you have suggestions on a fix for the bug -->
Well currently i do this: 
```kotlin
val polled = redisCommands.eval<String>("return redis.call('zpopmin', '$setKey', 1)[1]", ScriptOutputType.VALUE, arrayOf())
```

But that is just my workaround. 

#### Additional context

<!-- Add any other context about the problem here. Do not add code as screenshots. -->

Hey @nicolube ,

I am not able to reproduce the issue with the sample code you've provided.

> ... Caused by: java.lang.UnsupportedOperationException: io.lettuce.core.pubsub.PubSubCommandHandler$ResponseHeaderReplayOutput 

Iy also seems that you are using the connection for pub/sub? Can you provide a more complete example of your setup?

--- a/src/main/java/io/lettuce/core/output/ReplayOutput.java
+++ b/src/main/java/io/lettuce/core/output/ReplayOutput.java
@@ -34,6 +34,11 @@ public class ReplayOutput<K, V> extends CommandOutput<K, V, List<ReplayOutput.Si
     }
 
     @Override
+    public void set(double number) {
+        output.add(new Double(number));
+    }
+
+    @Override
     public void setError(ByteBuffer error) {
         error.mark();
         output.add(new ErrorBytes(error));
@@ -120,6 +125,21 @@ public class ReplayOutput<K, V> extends CommandOutput<K, V, List<ReplayOutput.Si
         final long message;
 
         Integer(long message) {
+            this.message = message;
+        }
+
+        @Override
+        protected void replay(CommandOutput<?, ?, ?> target) {
+            target.set(message);
+        }
+
+    }
+
+    static class Double extends Signal {
+
+        final double message;
+
+        Double(double message) {
             this.message = message;
         }
 

--- a/src/test/java/io/lettuce/core/pubsub/PubSubCommandIntegrationTests.java
+++ b/src/test/java/io/lettuce/core/pubsub/PubSubCommandIntegrationTests.java
@@ -28,6 +28,7 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.regex.Matcher;
@@ -541,6 +542,15 @@ class PubSubCommandIntegrationTests extends AbstractRedisClientTest {
         redis.spublish(shardChannel, shardMessage);
         assertThat(shardChannels.take()).isEqualTo(shardChannel);
         assertThat(messages.take()).isEqualTo(shardMessage);
+    }
+
+    @Test
+    void interleaveCommands() throws ExecutionException, InterruptedException {
+
+        // relay Double and Long
+        assertThat(pubsub.zadd("myzset", 1.0, "one").get()).isEqualTo(1L);
+        assertThat(pubsub.zadd("myzset", 2.0, "two").get()).isEqualTo(1L);
+        assertThat(pubsub.zpopmin("myzset", 1).get().get(0).getValue()).isEqualTo("one");
     }
 
     @Test

